src/ipv4.c

```diff
@@ -885,36 +885,46 @@ int ipv4_add_split_vpn_route(struct tunnel *tunnel, char *dest, char *mask,
 
 static int ipv4_set_split_routes(struct tunnel *tunnel)
 {
-	int i;
+	int i, j;
+	struct rtentry *gtw_rt = &tunnel->ipv4.gtw_rt;
 
 	for (i = 0; i < tunnel->ipv4.split_routes; i++) {
-		struct rtentry *route;
-		int ret;
+		struct rtentry *route = &tunnel->ipv4.split_rt[i];
+
+		for (j = 0; j < i ; j++) {
+			struct rtentry *other_route = &tunnel->ipv4.split_rt[j];
 
-		route = &tunnel->ipv4.split_rt[i];
-		// check if the route to be added is not the one to the gateway itself
-		if (route_dest(route).s_addr == route_dest(&tunnel->ipv4.gtw_rt).s_addr) {
-			log_debug("Skipping route to tunnel gateway (%s).\n",
-			          ipv4_show_route(route));
-			continue;
+			if (route_dest(route).s_addr == route_dest(other_route).s_addr)
+				break;
 		}
 
-		free(route_iface(route));
-		route_iface(route) = strdup(tunnel->ppp_iface);
-		if (!route_iface(route))
-			return ERR_IPV4_NO_MEM;
-		if (route_gtw(route).s_addr == tunnel->ipv4.ip_addr.s_addr)
-			route_gtw(route).s_addr = 0;
-		if (route_gtw(route).s_addr == 0)
-			route->rt_flags &= ~RTF_GATEWAY;
-		if (route_gtw(route).s_addr != 0)
-			route->rt_flags |= RTF_GATEWAY;
-		ret = ipv4_set_route(route);
-		if (ret == ERR_IPV4_SEE_ERRNO && errno == EEXIST)
-			log_warn("Route to gateway exists already.\n");
-		else if (ret != 0)
-			log_warn("Could not set route to tunnel gateway (%s).\n",
-			         err_ipv4_str(ret));
+		if (i == j) {
```

Comment:
That would result in less code modifications and light be more readable:
```c
		// skip duplicate routes
		if (j != i)
			continue
```

Reply:
Good point! Fixed!